security/rbac/scripts/enable-rbac-rest-proxy.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+
+################################################################################
+# Overview
+################################################################################
+#
+################################################################################
+
+# Source library
+. ../../../utils/helper.sh
+. ./rbac_lib.sh
+
+check_env || exit 1
+check_cli_v2 || exit 1
+check_jq || exit 1
+
+##################################################
+# Initialize
+##################################################
+
+. ../config/local-demo.cfg
+ORIGINAL_CONFIGS_DIR=/tmp/original_configs
+DELTA_CONFIGS_DIR=../delta_configs
+FILENAME=kafka-rest.properties
+create_temp_configs $CONFLUENT_HOME/etc/kafka-rest/$FILENAME $ORIGINAL_CONFIGS_DIR/$FILENAME $DELTA_CONFIGS_DIR/${FILENAME}.delta
+
+# Log in to Metadata Server (MDS)
+login_mds $MDS
+
+##################################################
+# Administrative Functions
+# - Start REST Proxy
+# - No additional role bindings are required because REST Proxy just does impersonation
+##################################################
+
+# Get the Kafka cluster id
+get_cluster_id_kafka
+
+confluent local start kafka-rest
+
+##################################################
+# REST Proxy client functions
+# - Try to view topics, before authorization (should see no topics)
+# - Grant the principal User:$CLIENTB to the ResourceOwner role for Topic:$TOPIC
+# - Try to view topics, after authorization (should see one topic $TOPIC)
+# - Create a consumer group $CONSUMER_GROUP
+# - Subscribe to the topic $TOPIC
+# - Consume messages from the topic $TOPIC, before authorization (should fail)
+# - Grant the principal User:$CLIENTB to the ResourceOwner role for Group:$CONSUMER_GROUP
+# - Consume messages from the topic $TOPIC, after authorization (should pass)
+##################################################
+
+TOPIC=test-topic-1
+
+echo -e "\n# Try to view topics, before authorization (should see no topics)"
+echo "curl -u clientb:clientb1 http://localhost:8082/topics"
+curl -u clientb:clientb1 http://localhost:8082/topics
+echo
+
+echo -e "\n# Grant the principal User:$CLIENTB to the ResourceOwner role for Topic:$TOPIC"
+echo "confluent iam rolebinding create --principal User:$CLIENTB --role ResourceOwner --resource Topic:$TOPIC --kafka-cluster-id $KAFKA_CLUSTER_ID"
+confluent iam rolebinding create --principal User:$CLIENTB --role ResourceOwner --resource Topic:$TOPIC --kafka-cluster-id $KAFKA_CLUSTER_ID
+
+echo -e "\n# Try to view topics, after authorization (should see one topic $TOPIC)"
+echo "curl -u clientb:clientb1 http://localhost:8082/topics"
+curl -u clientb:clientb1 http://localhost:8082/topics
+echo
+
+CONSUMER_GROUP=rest_proxy_consumer_group
+
+echo -e "\n# Create a consumer group $CONSUMER_GROUP"
```

Comment:
Should we add produce requests also?

Reply:
@omkreddy excellent suggestion, thanks.  Added this in latest commit